StitchCore/StitchCoreTests/Admin/StitchAdminClient.swift

```diff
@@ -0,0 +1,186 @@
+import Foundation
+import PromiseKit
+import ExtendedJson
+@testable import StitchCore
+
+// Any endpoint that can be described with basic
+// CRUD operations
+internal protocol Endpoint {
```

Comment:
I'm slightly okay with this protocol if all it was was a url property. `httpClient` seems like an odd coupling for describing something that conforms to an "endpoint"